### ISSUE TITLE
cloning of a VM which has both efidisk and tpmdisk from shared storage to loacal storage has issues

### ISSUE THREAD

**Issue by ChrisDek** [Telmate/terraform-provider-proxmox#1183]
For our clusters we use vms on a shared storage as templates to deploy them to the hosts using their local storage.

This works fine as long as there is no tpmdisk. All disks virtio0, (virtio1 if it exists),efidisk are cloned to the local storage.

When the template has efidisk and tpmdisk we end up with virtio0 on the local storage and efidiks and tpmdisk being created on the shared storage. Additionally we get an unused disk on the shared storage. Sadly this is not only cosmetic, but causes PVE´s snapshots function to fail.

Tested on 3.0.1-rc6 and PVE 8.2.7, host will be updated to 8.3 soon.

Anything else I should supply?

BR
Chris

**Comment by Tinyblargon** [comment]
@ChrisDek 

Find it a bit hard to follow. Could you give me 2 list, one of which storage the devices are supposed to end up and one where they are ending up?
